cf/data/data.py

```diff
@@ -8811,32 +8812,27 @@ def equals(
         other_Units = other.Units
         if self_Units != other_Units:
             logger.info(
-                "{}: Different Units ({!r}, {!r}".format(
+                "{}: Different Units ({!r}, {!r})".format(
                     self.__class__.__name__, self.Units, other.Units
                 )
             )
             return False
 
-        config = self.partition_configuration(readonly=True)
+        # other.to_memory()  # TODODASK is this still required?
 
-        other.to_memory()
+        self_dx = self._get_dask()
+        other_dx = other._get_dask()
 
-        for partition in self.partitions.matrix.flat:
-            partition.open(config)
-            array0 = partition.array
-            array1 = other[partition.indices].varray
-            partition.close()
-
-            if not _numpy_allclose(
-                array0, array1, rtol=float(rtol), atol=float(atol)
-            ):
-                logger.info(
-                    "{0}: Different array values (atol={1}, "
-                    "rtol={2})".format(self.__class__.__name__, atol, rtol)
-                )
+        # Finally check that corresponding elements are equal to a tolerance
+        if not da.allclose(
+            self_dx, other_dx, rtol=float(rtol), atol=float(atol)
+        ):
```

Comment:
(we would need a `.compute()` on the `da.allclose` call, but see previous comment!)